flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/stream/MiniBatchStreamingJoinOperator.java

```diff
@@ -0,0 +1,420 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.operators.join.stream;
+
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.util.RowDataUtil;
+import org.apache.flink.table.runtime.generated.GeneratedJoinCondition;
+import org.apache.flink.table.runtime.operators.bundle.trigger.BundleTriggerCallback;
+import org.apache.flink.table.runtime.operators.bundle.trigger.CoBundleTrigger;
+import org.apache.flink.table.runtime.operators.join.FlinkJoinType;
+import org.apache.flink.table.runtime.operators.join.stream.bundle.BufferBundle;
+import org.apache.flink.table.runtime.operators.join.stream.bundle.InputSideHasNoUniqueKeyBundle;
+import org.apache.flink.table.runtime.operators.join.stream.bundle.InputSideHasUniqueKeyBundle;
+import org.apache.flink.table.runtime.operators.join.stream.bundle.JoinKeyContainsUniqueKeyBundle;
+import org.apache.flink.table.runtime.operators.join.stream.state.JoinInputSideSpec;
+import org.apache.flink.table.runtime.operators.join.stream.state.JoinRecordStateView;
+import org.apache.flink.table.runtime.operators.metrics.SimpleGauge;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/** Streaming unbounded Join base operator which support mini-batch join. */
+public abstract class MiniBatchStreamingJoinOperator extends StreamingJoinOperator
+        implements BundleTriggerCallback {
+
+    private static final long serialVersionUID = -1106342589994963997L;
+
+    private final CoBundleTrigger<RowData, RowData> coBundleTrigger;
+
+    private transient BufferBundle<?> leftBuffer;
+    private transient BufferBundle<?> rightBuffer;
+    private transient SimpleGauge<Integer> leftBundleReducedSizeGauge;
+    private transient SimpleGauge<Integer> rightBundleReducedSizeGauge;
+
+    public MiniBatchStreamingJoinOperator(MiniBatchStreamingJoinParameter parameter) {
+        super(
+                parameter.leftType,
+                parameter.rightType,
+                parameter.generatedJoinCondition,
+                parameter.leftInputSideSpec,
+                parameter.rightInputSideSpec,
+                parameter.leftIsOuter,
+                parameter.rightIsOuter,
+                parameter.filterNullKeys,
+                parameter.leftStateRetentionTime,
+                parameter.rightStateRetentionTime);
+
+        this.coBundleTrigger = parameter.coBundleTrigger;
+    }
+
+    @Override
+    public void open() throws Exception {
+        super.open();
+
+        // must copy in mini-batch
+        this.requiresCopy = true;
+
+        this.leftBuffer = initialBuffer(leftInputSideSpec);
+        this.rightBuffer = initialBuffer(rightInputSideSpec);
```

Comment:
this.leftSerializer = leftType.createSerializer(new ExecutionConfig());
        this.rightSerializer = rightType.createSerializer(new ExecutionConfig());

Reply:
Yes, I have changed this and simplified the code.